core/src/main/scala-3/caliban/introspection/IntrospectionDerivation.scala

```diff
@@ -5,14 +5,24 @@ import caliban.Value.StringValue
 import caliban.introspection.adt._
 import caliban.parsing.adt.Directive
 import caliban.schema.Schema
-import caliban.schema.Schema.auto._
-import caliban.schema.ArgBuilder.auto._
+import caliban.schema.ArgBuilder
 
 trait IntrospectionDerivation {
-  import Schema._
+  private given Schema[Any, __InputValue]     = Schema.derived
+  private given Schema[Any, __EnumValue]      = Schema.derived
+  private given Schema[Any, __Field]          = Schema.derived
+  private given Schema[Any, __Type]           = Schema.derived
+  private given Schema[Any, __TypeArgs]       = Schema.derived
+  private given Schema[Any, __Schema]         = Schema.derived
+  private given Schema[Any, __Directive]      = Schema.derived
+  private given Schema[Any, __DeprecatedArgs] = Schema.derived
 
-  implicit lazy val inputValueSchema: Schema[Any, InputValue]   = Schema.gen
-  implicit lazy val typeSchema: Schema[Any, __Type]             = Schema.gen
-  implicit lazy val __directiveSchema: Schema[Any, __Directive] = Schema.gen
-  val introspectionSchema: Schema[Any, __Introspection]         = Schema.gen
+  // Unions, so we can auto-derive them cheaply
+  private given Schema[Any, __TypeKind]          = Schema.Auto.derived
+  private given Schema[Any, __DirectiveLocation] = Schema.Auto.derived
```

Comment:
Love that we can do this in Scala 3. It was only by accident that I discovered it was possible without adding an import